### PR TITLE
feat: Add support for HONEYCOMB_API_KEY environment variable for provider

### PR DESCRIPTION
**Pull request by teddylear**
feat: Add support for HONEYCOMB_API_KEY environment variable for provider

@jharley Converted current tests to use new environment variable and updated documentation saying that both are supported, but new env var takes priority. If there's anything I should change or test please let me know. 

Also I count not get the tests working locally with my API key, does it required an enterprise account for SLO tests? 

Addresses #117

### CODE REVIEW

**Review by jharley — CHANGES_REQUESTED** [review]
Hey, @teddylear! Thanks for picking this up.

This is a great, but the intended variable name is `HONEYCOMB_API_KEY` (to match many of our other integrations) and you've got `HONEYCOMBIO_API_KEY`. If you could correct that we can get this in (and I'll add the new variable to CI)

**Comment by teddylear** [comment]
Ahh Sorry I missed that, I can update.

**Comment by teddylear** [comment]
@jharley Updated to use correct env var name now, please let me know if there's anything else I should test/add/change. Also, thanks for the quick initial review!

File: honeycombio/provider.go [hunk 1]
```diff
@@ -25,10 +25,13 @@ func Provider() *schema.Provider {
...
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("HONEYCOMBIO_APIKEY", nil),
-				Sensitive:   true,
+				Type:     schema.TypeString,
+				Required: true,
+				// Tiering goes HONEYCOMB_API_KEY > HONEYCOMBIO_APIKEY > nil
+				DefaultFunc: schema.EnvDefaultFunc(
+					"HONEYCOMB_API_KEY", schema.EnvDefaultFunc(
+						"HONEYCOMBIO_APIKEY", nil)),
```
Comment (jharley) [inline honeycombio/provider.go:33]: I'd like to see a call to `tflog.Warn()` here to notify people that they're using the deprecated env var. What do you think?

Comment (teddylear) [reply honeycombio/provider.go:33]: I like that idea, let me add it.

Comment (teddylear) [reply honeycombio/provider.go]: @jharley Ok added change, please let me know if there's anything else to update or change.

File: honeycombio/provider.go [hunk 2]
```diff
@@ -21,14 +22,25 @@ func init() {
 // overwritten during the release process.
 var providerVersion = "dev"
 
+func deprecatedEnvCheck() schema.SchemaDefaultFunc {
+	if _, ok := os.LookupEnv("HONEYCOMBIO_APIKEY"); ok {
+		ctx := context.Background()
+		tflog.Warn(ctx, "HONEYCOMBIO_APIKEY is being depcrated, please use HONEYCOMB_API_KEY instead")
```
Comment (jharley) [inline honeycombio/provider.go:28]: Minor typo/phrasing fix, but this is great! ✨ 

```suggestion
		tflog.Warn(ctx, "HONEYCOMBIO_APIKEY has been deprecated, please use HONEYCOMB_API_KEY instead")
```

Comment (teddylear) [reply honeycombio/provider.go:28]: Ahhh let me make change.

Comment (jharley) [reply honeycombio/provider.go:28]: Thank you! We'll get this into 0.9.0

**Comment by teddylear** [comment]
@jharley Committed your suggested change, let me know if there's anything else I should change.

**Review by jharley — APPROVED** [review]
Lookin' good! Thanks for this, @teddylear 🎉

**Pull request merged**